frontend/src/components/layout/expected-origin-boundary.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { headers } from 'next/headers'
+import type { PropsWithChildren } from 'react'
+import React from 'react'
+
+export interface ExpectedOriginBoundaryProps extends PropsWithChildren {
+  expectedOrigin: string
+}
+
+export const buildOriginFromHeaders = (): string | undefined => {
+  const headers1 = headers()
```

Comment:
Variable name isn't ideal.
And yes I know this is because the conflicting import.